packages/core/ios/RNEmbrace/EmbraceManager.swift

```diff
@@ -20,565 +20,599 @@ private let EVENT_TIMESTAMP_KEY = "timeStampMs"
+    rejecter reject: @escaping RCTPromiseRejectBlock
+  ) {
     
-    @objc
-    func getDeviceId(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
-        if let deviceId = Embrace.client?.currentDeviceId() {
-            resolve(deviceId)
-        } else {
-            reject("GET_DEVICE_ID", "Error getting deviceId", nil)
-        }
-    }
+    let severityValue = self.severityFromString(from:severity)
+    guard let attributes = properties as? [String: String] else {
+      reject("LOG_MESSAGE_INVALID_PROPERTIES", "Properties should be [String: String]", nil)
+      return
+    }
+    Embrace.client?.log(
+      message,
+      severity: severityValue,
+      attributes: attributes
+    )
+    resolve(true)
     
-    @objc(setUserIdentifier:resolver:rejecter:)
-    func setUserIdentifier(_ userIdentifier: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
-        Embrace.client?.metadata.userIdentifier = userIdentifier
-        resolve(true)
-    }
+  }
+  
+  @objc(startView:resolver:rejecter:)
+  func startView(_ viewName: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    let startTimeDate = Date()
     
+    let span = Embrace.client?.buildSpan(name: "emb-screen-view")
+      .setAttribute(key:"view.name", value:viewName)
+      .setAttribute(key:"emb.type", value:"ux.view")
+      .setStartTime(time: startTimeDate)
+      .startSpan()
     
-    @objc
-    func getCurrentSessionId(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
-        if let sessionId = Embrace.client?.currentSessionId() {
-            resolve(sessionId)
-        } else {
-            reject("GET_SESSION_ID", "Error getting sessionId", nil)
-        }
+    var spanId = ""
+    if span != nil {
+      spanId = spanRepository.spanStarted(span: span!)
     }
     
-    @objc(addBreadcrumb:resolver:rejecter:)
-    func addBreadcrumb(_ event: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
-        //TODO Refactor SPAN
-        //      do {
-        //        try Embrace.client?.add(event: event)
-        //          resolve(true)
-        //      } catch {
-        //          reject(false)
-        //      }
+    if spanId.isEmpty {
+      reject("START_SPAN_ERROR", "Failed to start span", nil)
+    } else {
+      resolve(spanId)
+    }
+  }
+  
+  @objc(endView:resolver:rejecter:)
+  func endView(_ spanId: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    let endTimeDate = Date()
+    let endTimeMS = NSNumber(value: endTimeDate.timeIntervalSince1970 * 1000)
+    
+    stopSpan(spanId, errorCodeString: "", endTimeMs: endTimeMS, resolver: resolve, rejecter: reject)
```

Comment:
shouldn't need to pass in end time, should default to current time if given a 0
```suggestion
    stopSpan(spanId, errorCodeString: "", endTimeMs: 0.0, resolver: resolve, rejecter: reject)
```